### DISCUSSION TITLE
Overlapping opaque black polygons with white edges

### DISCUSSION THREAD

**Discussion by superfede76** [mozman/ezdxf#1030] [Q&A]
Hi all,
I need to overlap some triangles so that you can only see the white edge, and inside it is black. If I try to fill the interior with another color,fro example blue, it works, but with black it does not.

this is my code working with blu tringles with white boundary

```python
import ezdxf
import random

def create_solid_triangles_inside_square(num_triangles):
    doc = ezdxf.new(dxfversion="R2010")
    msp = doc.modelspace()
    for _ in range(num_triangles):
        triangle_vertices = [
            (random.uniform(0, 10), random.uniform(0, 10)),
            (random.uniform(0, 10), random.uniform(0, 10)),
            (random.uniform(0, 10), random.uniform(0, 10))
        ]
        msp.add_solid(triangle_vertices, dxfattribs={'color': 1})
        msp.add_lwpolyline(triangle_vertices, close=True, dxfattribs={'color': 7})
    doc.saveas("marker.dxf")

if __name__ == "__main__":
    num_triangles = 5
    create_solid_triangles_inside_square(num_triangles)

```

 Can anyone help?
Thank you

**Edit**: I need to overlap some triangles so that you can only see the white edge, and inside it is **empty** (not black!) like in the below example.

![result](https://github.com/mozman/ezdxf/assets/7916852/e79ed6bf-fc82-4324-af15-909f0fffe201)

**Comment by mozman** [comment]
You cannot use ACI as color for that specific use case. ACI stands for AutoCAD Color Index and is simply an index into a vendor specific color table or a color table that you provide as a `.ctb` file. The index 7 is very special because it is white on a dark background and black on a light background.  You could use ACI 250 for black and 255 for white:

DWG Trueview Modelspace:

![image](https://github.com/mozman/ezdxf/assets/4230966/d3af660f-cd56-4022-954c-d9facffedf54)

DWG Trueview Paperspace:

![image](https://github.com/mozman/ezdxf/assets/4230966/3794a75a-d04d-4d6d-b720-46ce088eb2ee)

But BricsCAD disagree about these default colors:

![image](https://github.com/mozman/ezdxf/assets/4230966/0fda32c0-6a84-49f9-99d2-bb5b55b6a266)

![image](https://github.com/mozman/ezdxf/assets/4230966/9c5e03ed-62e0-4d4b-a05a-9e22e7464a78)

The better way is to use true color values:
```python
import ezdxf
from ezdxf import colors
import random


def create_solid_triangles_inside_square(num_triangles):
    doc = ezdxf.new(dxfversion="R2010")
    msp = doc.modelspace()
    for _ in range(num_triangles):
        triangle_vertices = [
            (random.uniform(0, 10), random.uniform(0, 10)),
            (random.uniform(0, 10), random.uniform(0, 10)),
            (random.uniform(0, 10), random.uniform(0, 10)),
        ]
        msp.add_solid(
            triangle_vertices, dxfattribs={"true_color": colors.rgb2int((0, 0, 0))}
        )
        msp.add_lwpolyline(
            triangle_vertices,
            close=True,
            dxfattribs={"true_color": colors.rgb2int((255, 255, 255))},
        )
    doc.saveas("marker_rgb.dxf")


if __name__ == "__main__":
    num_triangles = 5
    create_solid_triangles_inside_square(num_triangles)
```

That guarantees to get the same color in every CAD application (if you ignore color spaces):

![image](https://github.com/mozman/ezdxf/assets/4230966/627983bf-c8ff-43bc-8248-f332f24095d7)

![image](https://github.com/mozman/ezdxf/assets/4230966/7871bf39-7d9f-4661-986e-f28617a38568)

![image](https://github.com/mozman/ezdxf/assets/4230966/1e579a67-aa2a-4491-98a3-d2c0f2af020b)

![image](https://github.com/mozman/ezdxf/assets/4230966/30951437-a68c-428a-b442-402d597dff77)

**Reply by superfede76** [reply]
Hi Mozamn, thanks for the reply. I realized that I am explained badly. What I want to do is erase everything that's inside the triangle and don't color it in black. Here an example:
​
![result](https://github.com/mozman/ezdxf/assets/7916852/ff93b0f4-0d77-4856-9195-53b877102449)
